Update G195 - removal of last line of test (low contrast OK if focus ring > 2px thick)

detlevhfischer opened the pull request:
Deletion of last line of test procedure that indicated that contrast below 3:1 is OK if the focus ring is more than 2px thick

Closes https://github.com/w3c/wcag/issues/3682

detlevhfischer committed:
Update G195.html

Deletion of last line of test procedure that indicated that contrast below 3:1 is OK if the border is more than 2px thick

bruce-usab commented:
Discussed on TF call 11/8 and moved to Ready for approval after Patrick offered to double check.

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

mbgower commented:
@patrickhlauke and @detlevhfischer please see my comments in the issue. I believe we need to decide whether we are to maintain two versions of G195. Either way, we need more changes in the Test (unless we are intentionally exceeding the Non-text Contrast requirements). Also, if we decide _not_ to apply to 2.4.13 in 2.2, we need to make adjustments to the Understanding document to remove the mapping.

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

mbgower commented:
Discussed on the TF admin call.
I will do the following:
- rewrite the G195 test to be specific to meet Non-text Contrast
- investigate other techniques that attempt to meet both AA and AAA (e.g., G17 and G18)
- draft a new technique to meet Focus Appearance
- Consider the title and guidance in G195 and the new technique to ensure they are complementary but also distinctive

alastc commented:
I agree with having two separate techniques:

- G195 is for WCAG 2.1, and we remove the focus-appearance aspects.
- G2xx for WCAG 2.2, which meets the focus-appearance one. Include a note at the top saying something like "This builds on G195 to cover Focus-Appearance". 

This PR could be just for the first part, and separate the second one.

mbgower committed:
Update focus-appearance.html

Remove cross-reference to G195 from AAA SC

mbgower committed:
Update focus-appearance.html

Putting the pointer to G195 back into WCAG 2.2, as per my latest comment in the issue https://github.com/w3c/wcag/issues/3682#issuecomment-2504785223

mbgower committed:
Update G195.html

Updating language of test to meet wording of 2.4.13 Focus Appearance

mbgower committed:
Update G195.html

mbgower committed:
Update G195.html

Modifying the second check to add in a contrast requirement to meet 1.4.11.

mbgower committed:
Update G195.html

formatting

mbgower commented:
@alastc as per the comments in the issue, I found that if we can do it, it actually makes more sense to extend the test to meet Focus Appearance. If necessary, we can sever the connections from 2.1, but I have now made the test include a check that it meets 1.4.11 requirements, so even though the technique exceeds 1.4.11, it could still stay as being sufficient.

mbgower committed:
Update G195.html

Small changes to make the text match the example on the page as well as the test criteria.

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

mbgower commented on techniques/general/G195.html:
```diff
@@ -19,23 +19,25 @@ <h3>Form Elements</h3>
       <section class="results"><h3>Expected Results</h3>
          <ul>
-            <li>Checks #2, #4, and #5 are true.</li>
+            <li>If Check #1 is true, all other checks are also true.</li>
```
```suggestion
            <li>If Check #1 is true, all other checks must also be true.</li>
```

mbgower commented:
@mbgower to merge the first to test steps into the preamble
For each user interface component on the page that should receive keyboard focus:</p>
         <ol>
            <li>Check that the focus indicator or its background color has been modifed by the author</li>

mbgower committed:
Update G195.html

Updated test based on TF discussion

bruce-usab commented:
Discussed on TF call 12-6, facilitators to discuss on a coordination call.

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

patrickhlauke committed:
Fix broken markup, remove empty lines

patrickhlauke commented on techniques/general/G195.html:
```diff
@@ -2,7 +2,7 @@
       <p>Generally applicable.</p>
    </section><section id="description"><h2>Description</h2>
       <p>The objective of this technique is enhance the focus indicator in the browser, by creating a visible one in the content. The default focus indicator in some browsers is a thin, dotted, black line. It can be difficult to see the line when it is around a form element which already has an outline, when the focused element is inside a table cell, when the focused element is very small, or wh ...
```
```suggestion
      <p>The objective of this technique is to enhance the focus indicator in the browser, by creating a visible one in the content. The default focus indicator in some browsers is a thin, dotted, black line. It can be difficult to see the line when it is around a form element which already has an outline, when the focused element is inside a table cell, when the focused element is very small, or when the background of the page is a dark color. Some browsers use a pale blue outline, which can be difficult to see on some backgrounds.</p>
```

patrickhlauke committed:
Update techniques/general/G195.html

bruce-usab commented:
Discussed on backlog call 1/17.  Needs another round of editorial review.  Expected Results might mention relationship to other SC.  Patrick volunteered to take another pass.

patrickhlauke committed:
Added minor clarification to test steps about which SC these satisfy

patrickhlauke committed:
Merge branch 'DF-modify-test-pocedure' of https://github.com/w3c/wcag into DF-modify-test-pocedure

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure

patrickhlauke committed:
Merge branch 'main' into DF-modify-test-pocedure
